core/src/main/java/org/infinispan/configuration/cache/BackupConfigurationBuilder.java

```diff
@@ -148,6 +148,9 @@ public void validate() {
          throw new CacheConfigurationException("It is required to specify a 'failurePolicyClass' when using a " +
                                                 "custom backup failure policy!");
       }
+      if (useTwoPhaseCommit && strategy == BackupConfiguration.BackupStrategy.ASYNC) {
+         throw new CacheConfigurationException("'two-phase-commit' can be used only with SYNC backup strategy!");
```

Comment:
IMO, they should be in the ```Log``` interface, but no other ```*ConfigurationBuilder``` is using it. 
So, it is better to leave them (the old an the new) as it is to keep them consistent.